### PR TITLE
ci: separate e2e and backend ci flows

### PR DESCRIPTION
**Pull request by wanlingt**
## Problem

The e2e and backend tests were combined in #5272 so that they can both use `npm run build`, however they take a long time to run.

## Solution

Save the build files as an artifact so that they can be used later in the separated e2e and backend tests

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (justynoh) [inline .github/workflows/ci.yml]: we should actually shard out the build into it's own step right? since they're common between both backend and e2e, then they can share on the build

Comment (karrui) [reply .github/workflows/ci.yml]: yarh

Comment (wanlingt) [reply .github/workflows/ci.yml]: yeah context is this is what we had initially, but the files from the build step weren't being used in e2e and backend tests so we combined the build, e2e and backend steps tgt. discussed w @karrui - can use a build artifact

**Comment by wanlingt** [comment]
note to self: update node version to 18

**Comment by wanlingt** [comment]
@justynoh @karrui successful ci [here](https://github.com/opengovsg/FormSG/actions/runs/3598169928)

will remove the `build_backend_and_e2e_test` requirement for merging PRs when this is approved, and add the build, backend, e2e requirement for merging PRs separately in github settings

i've also opened a separate [issue](https://github.com/opengovsg/FormSG/issues/5512) for upgrading the workflows to use node version 18 because i'm having some trouble with NODE_OPTIONS and want to merge this in first haha

**Comment by karrui** [comment]
Wondering whether we can rerun tests with this flow if the build artefacts disappear too fast. May need some more investigation.

**Comment by wanlingt** [comment]
> Wondering whether we can rerun tests with this flow if the build artefacts disappear too fast. May need some more investigation.

good point! I'll just remove the retention policy https://github.com/opengovsg/FormSG/pull/5483/commits/68512e4e6ad60854850ea88b98573c172e7ae348
from [docs](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts): GitHub stores artifacts for 90 days, and is free for public repos

(will reinstate the conditionals https://github.com/opengovsg/FormSG/pull/5483/commits/eedc5f407c31fd14dcdee96ba2b783d248b37645 after this PR is approved)

File: .github/workflows/ci.yml
Comment (karrui) [inline .github/workflows/ci.yml]: supernit: new line

File: .github/workflows/ci.yml
Comment (karrui) [inline .github/workflows/ci.yml]: should we reenable this check too? it should only run if there are backend changes imo

File: .github/workflows/ci.yml
Comment (karrui) [inline .github/workflows/ci.yml]: same

File: .github/workflows/ci.yml
Comment (karrui) [inline .github/workflows/ci.yml:62]: we can shunbian upgrade this node version to 18 liow

File: .github/workflows/ci.yml
Comment (karrui) [inline .github/workflows/ci.yml:119]: Same (also for the rest. Actually may be in a different PR hor)

**Review by karrui — APPROVED** [review]
lgtm if the stuff commented are done in another PR

File: .github/workflows/ci.yml
Comment (wanlingt) [reply .github/workflows/ci.yml]: yup will reenable after it's approved! wanted to show that the separate flows run successfully

File: .github/workflows/ci.yml
Comment (wanlingt) [reply .github/workflows/ci.yml:119]: ya correct separate pr!

File: .github/workflows/ci.yml
Comment (wanlingt) [reply .github/workflows/ci.yml]: [af93bca](https://github.com/opengovsg/FormSG/pull/5483/commits/af93bcac96b49e9550eea3074776d4f226fdfdbf)

File: .github/workflows/ci.yml
Comment (wanlingt) [reply .github/workflows/ci.yml]: [af93bca](https://github.com/opengovsg/FormSG/pull/5483/commits/af93bcac96b49e9550eea3074776d4f226fdfdbf)

**Review by justynoh — APPROVED** [review]
lgtm

**Pull request merged**
